### ISSUE TITLE
Inline images based MIME type, not file extension

### ISSUE THREAD

**Issue by kensanata** [skyjake/lagrange#373]
Currently Lagrange appears to inline images if the extension suggests an image, e.g. .jpg; since Gemini (like the web) allows resources to provide an independent MIME type, I often upload images without extensions. I think it'd be great if Lagrange handled these situations. Sure, at first it looks like an ordinary link to be followed, but as soon as MIME type is known, Lagrange could return to the previous page and inline the image.

Inlining works here:
gemini://alexschroeder.ch/do/gallery/2020-aminona

This is a test page:
gemini://transjovian.org/test/page/testing

It links to an image as follows:

```
Linking an image without a typical file extension:
=> /test/file/Alex%E2%80%99s%20Avatar
```

**Comment by skyjake** [comment]
This has been a nice-to-have on my todo list ever since the beginning. In the wild, though, extensionless image links seem rare so it hasn't been implemented.

When a new request begins, the old page isn't discarded until the new request's media type is known. At that point the decision could be made to inline the results under the clicked link. There's a bit of bookkeeping in `DocumentWidget` that's unique to inline media requests, but setting that up for an ongoing `GmRequest` is pretty trivial.

**Comment by skyjake** [comment]
Implemented for v1.10.
